cachito/workers/pkg_managers/general_js.py

```diff
@@ -118,23 +152,40 @@ def download_dependencies(request_id, deps, proxy_repo_url, skip_deps=None):
 
             if counter % batch_size == 0:
                 deps_batches.append([])
-            deps_batches[-1].append(dep_identifier)
+            deps_batches[-1].append((dep_identifier, external_dep_version))
             downloaded_deps.add(dep_identifier)
             counter += 1
 
         for dep_batch in deps_batches:
-            log.debug(f"Downloading the following npm dependencies: {', '.join(dep_batch)}")
-            npm_pack_args = ["npm", "pack"] + dep_batch
+            # Create a list of dependencies to be downloaded. Excluding 'external_dep_version'
+            # from the list of tuples
+            dep_batch_download = [i[0] for i in dep_batch]
+            log.debug(
+                f"Downloading the following npm dependencies: {', '.join(dep_batch_download)}"
+            )
+            npm_pack_args = ["npm", "pack"] + dep_batch_download
             output = run_cmd(npm_pack_args, run_params, "Failed to download the npm dependencies")
 
             # Move dependencies to their respective folders
             # Iterate through the tuples made of dependency tarball and dep_identifier
             # e.g. ('angular-animations-8.2.0.tgz', '@angular/animations@8.2.0')
             for dep_pair in list(zip(output.split("\n"), dep_batch)):
                 tarball = dep_pair[0]  # e.g. angular-animations-8.2.0.tgz
-                dir_path = dep_pair[1].rsplit("@", 1)[0]  # e.g. @angular/animations
+                dir_path = dep_pair[1][0].rsplit("@", 1)[0]  # e.g. @angular/animations
+                external_dep_version = dep_pair[1][1]
+
+                # In case of external dependencies, create additional intermediate
+                # parent e.g. github/<org>/<repo> or external-<repo>
+                extra_dir = ""
+                if external_dep_version is not None:
+                    match = re.match(r"(?P<prefix>[^:]+):?(?P<suffix>[^/]+)", external_dep_version)
+                    if match.group("prefix") in known_git_prefixes:
+                        extra_dir = os.path.join(match.group("prefix"), match.group("suffix"))
+                    elif match.group("prefix") in other_prefixes:
+                        dir_path = f"external-{dir_path}"
+
```

Comment:
I'm thinking this might be simpler to read and less fragile and then you wouldn't need to consult `known_git_prefixes` or `other_prefixes`.
```suggestion
                # In case of external dependencies, create additional intermediate
                # parent e.g. github/<namespace>/<repo> or external-<repo>
                external_dep_version = dep_pair[1][1]
                if external_dep_version:
                    match = re.match(
                        r"^(?P<host>.+)(?::)(?P<namespace>[^/]+)(?:\/)(?P<repo>[^/#]+)(?:.+)$",
                        external_dep_version,
                    )
                    if match:
                        groups = match.groupdict()
                        dir_path = os.path.join(groups["host"], groups["namespace"], groups["repo"])
                    else:
                        dir_path = f"external-{dir_path}"
                else:
                    dir_path = dep_pair[1][0].rsplit("@", 1)[0]  # e.g. @angular/animations
```